app/components/channel_drawer_list/channel_drawer_list.js

```diff
@@ -326,90 +312,91 @@ class ChannelDrawerList extends Component {
-            directNonTeamChannels
+            directAndGroupChannels
         } = props.channels;
 
         if (favoriteChannels.length) {
             data.push(
-                <FormattedText
-                    style={[Styles.title, {color: theme.sidebarText}]}
-                    id='sidebar.favorite'
-                    defaultMessage='FAVORITES'
-                />,
+                this.renderTitle(styles, 'sidebar.favorite', 'FAVORITES', null, favoriteChannels.length > 0),
                 ...favoriteChannels
             );
         }
 
         data.push(
-            <View style={{flex: 1, flexDirection: 'row', alignItems: 'center'}}>
-                <FormattedText
-                    style={[Styles.title, {color: theme.sidebarText}]}
-                    id='sidebar.channels'
-                    defaultMessage='CHANNELS'
-                />
-                {this.renderSectionAction(this.props.actions.showMoreChannelsModal)}
-            </View>,
+            this.renderTitle(styles, 'sidebar.channels', 'CHANNELS', this.props.actions.showMoreChannelsModal, publicChannels.length > 0),
             ...publicChannels
         );
 
         data.push(
-            <View style={{flex: 1, flexDirection: 'row', alignItems: 'center'}}>
-                <FormattedText
-                    style={[Styles.title, {color: theme.sidebarText}]}
-                    id='sidebar.pg'
-                    defaultMessage='PRIVATE GROUPS'
-                />
-                {this.renderSectionAction(this.createPrivateChannel)}
-            </View>,
+            this.renderTitle(styles, 'sidebar.pg', 'PRIVATE GROUPS', this.createPrivateChannel, privateChannels.length > 0),
             ...privateChannels
         );
 
         data.push(
-            <View style={{flex: 1, flexDirection: 'row', alignItems: 'center'}}>
-                <FormattedText
-                    style={[Styles.title, {color: theme.sidebarText}]}
-                    id='sidebar.direct'
-                    defaultMessage='DIRECT MESSAGES'
-                />
-                {this.renderSectionAction(this.props.actions.showDirectMessagesModal)}
-            </View>,
-            ...directChannels
+            this.renderTitle(styles, 'sidebar.direct', 'DIRECT MESSAGES', this.props.actions.showDirectMessagesModal, directAndGroupChannels.length > 0),
+            ...directAndGroupChannels
         );
 
-        if (directNonTeamChannels.length) {
-            data.push(
-                <LineDivider
-                    color={theme.sidebarTextActiveBorder}
-                    translationId='sidebar.otherMembers'
-                    translationText='Outside this team'
-                />,
-                ...directNonTeamChannels
-            );
-        }
-
         this.firstUnreadChannel = null;
         this.lastUnreadChannel = null;
         this.findUnreadChannels(data);
 
         return data;
     };
 
+    renderSectionAction = (styles, action) => {
```

Comment:
I'm thinking more for encapsulation and to make it a bit more obvious what each of the 5 parameters for renderTitle are. But as you said, they won't be reused so it doesn't really matter